pkg/cluster/cluster.go

```diff
@@ -35,23 +35,62 @@ type Peer struct {
 	stopc chan struct{}
 }
 
+var (
+	pushPullInterval = 5 * time.Second
+	gossipInterval   = 5 * time.Second
+)
+
 // PeerType describes a peer's role in the cluster.
 type PeerType string
 
 // Constants holding valid PeerType values.
 const (
-	PeerTypeStore = "store"
-	PeerTypeQuery = "query"
+	PeerTypeStoreGCS     = "store-gcs"
+	PeerTypeStoreSidecar = "store-sidecar"
+	PeerTypeQuery        = "query"
 )
 
+// PeerState contains state for the peer.
+type PeerState struct {
+	Type    PeerType
+	APIAddr string
+
+	Metadata PeerMetadata
+}
+
+// PeerMetadata are the information that can change in runtime of the peer.
+type PeerMetadata struct {
+	Labels []storepb.Label
+
+	// LowTimestamp indicates the minTime of the oldest block available from this peer.
+	LowTimestamp int64
+	// HighTimestamp indicates the maxTime of the youngest block available from this peer.
+	HighTimestamp int64
+}
+
+// MetadataUpdater is a function that allows to update metadata of the peer.
+type MetadataUpdater interface {
+	SetLabels([]storepb.Label)
+	SetTimestamps(lowTimestamp int64, highTimestamp int64)
```

Comment:
In general, maybe this is not an ideal place for the type. If I see it correctly it's meant to decouple `pkg/shipper` and `pkg/store` from the concrete implementation. But since this interface still lives in `pkg/custer` there's a package dependency regardless.

I might be nit-picky here but the more we avoid those, the more likely we'll end up with a coherent and decoupled design at the end.

Reply:
So this is basically because of two reasons:
1) Not putting whole `*Peer ` into shipper and store.. We want to give the methods which are only needed + make it single functionality if possible.
2) We can mock it easily for testing purposes